Sources/SKCore/BuildServerBuildSystem.swift

```diff
@@ -117,14 +117,20 @@ public final class BuildServerBuildSystem {
   }
 
   private func initializeBuildServer() throws {
-    guard let interpreter =
-        lookupExecutablePath(filename: executable("python3"),
-                             searchPaths: searchPaths) ??
-        lookupExecutablePath(filename: executable("python"),
-                             searchPaths: searchPaths) else {
-      throw BuildServerTestError.executableNotFound("python3")
+    var serverPath = AbsolutePath(serverConfig.argv[0], relativeTo: projectRoot)
+    var flags = Array(serverConfig.argv[1...])
+    if serverPath.suffix == "py" {
+      flags = [serverPath.pathString] + flags
+      guard let interpreterPath =
+          lookupExecutablePath(filename: executable("python3"),
+                               searchPaths: searchPaths) ??
+          lookupExecutablePath(filename: executable("python"),
+                               searchPaths: searchPaths) else {
+        throw BuildServerTestError.executableNotFound("python3")
+      }
+
+      serverPath = interpreterPath
     }
```

Comment:
Oh, I didn't realize that this was in SKCore, this makes sense.  I thought that this was a test-only method, which would be fine to assume that the build server would never be an executable.